src/Microsoft.VisualStudio.Web.CodeGeneration.Core/FileSystemChangeTracker.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.VisualStudio.Web.CodeGeneration.Contracts.FileSystemChange;
+
+namespace Microsoft.VisualStudio.Web.CodeGeneration
+{
+    public class FileSystemChangeTracker : IFileSystemChangeTracker
+    {
+        private object _syncobject = new object();
+
+        private Dictionary<string, FileSystemChangeInformation> _changes = new Dictionary<string, FileSystemChangeInformation>();
+        public IEnumerable<FileSystemChangeInformation> Changes
+        {
+            get
+            {
+                lock (_syncobject)
+                {
+                    return _changes.Values;
```

Comment:
You'll probably want to copy the value to another collection inside the lock, otherwise the mutable collection is still being shared after the lock has been released